apps/judicial-system/backend/migrations/20221107150202-update-notification.js

```diff
@@ -0,0 +1,103 @@
+'use strict'
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    return queryInterface.sequelize.transaction(async (transaction) =>
+      // temp column for migratating recipients
+      queryInterface
+        .addColumn(
+          'notification',
+          'recipients_json',
+          {
+            defaultValue: [],
+            type: Sequelize.ARRAY(Sequelize.JSON),
+            allowNull: false,
+          },
+          { transaction },
+        )
+        .then(() =>
+          // get all notifications and migrate recipients to recipients_json
+          queryInterface.sequelize
+            .query(`select * from notification`, { transaction }))
+        .then(async ([notifications]) => {
+          for (const notification of notifications) {
+            const fallback = [{ success: false }]
+
+            let parsedRecipients = JSON.parse(notification.recipients)
+            if (!parsedRecipients || parsedRecipients.length === 0) {
+              parsedRecipients = fallback
+            }
+
+            await queryInterface.bulkUpdate(
```

Comment:
If we don't wait for these updates, we will remove the old column and rename the new column to `recipients`. It is probably fine since this is all in a transaction and you already have all the data in the old table in memory.

Reply:
The old table and data is also in the database until you commit the transaction. The database server takes care of this completely and don't even need to think about it if we use transactions.